.settings/org.eclipse.jdt.core.prefs

```diff
@@ -387,14 +454,37 @@ org.eclipse.jdt.core.formatter.insert_space_between_empty_parens_in_method_decla
 org.eclipse.jdt.core.formatter.insert_space_between_empty_parens_in_method_invocation=do not insert
 org.eclipse.jdt.core.formatter.join_lines_in_comments=true
 org.eclipse.jdt.core.formatter.join_wrapped_lines=false
+org.eclipse.jdt.core.formatter.keep_annotation_declaration_on_one_line=one_line_never
+org.eclipse.jdt.core.formatter.keep_anonymous_type_declaration_on_one_line=one_line_never
+org.eclipse.jdt.core.formatter.keep_code_block_on_one_line=one_line_never
 org.eclipse.jdt.core.formatter.keep_else_statement_on_same_line=false
 org.eclipse.jdt.core.formatter.keep_empty_array_initializer_on_one_line=false
+org.eclipse.jdt.core.formatter.keep_enum_constant_declaration_on_one_line=one_line_never
+org.eclipse.jdt.core.formatter.keep_enum_declaration_on_one_line=one_line_never
+org.eclipse.jdt.core.formatter.keep_if_then_body_block_on_one_line=one_line_never
 org.eclipse.jdt.core.formatter.keep_imple_if_on_one_line=false
+org.eclipse.jdt.core.formatter.keep_lambda_body_block_on_one_line=one_line_never
+org.eclipse.jdt.core.formatter.keep_loop_body_block_on_one_line=one_line_never
+org.eclipse.jdt.core.formatter.keep_method_body_on_one_line=one_line_never
+org.eclipse.jdt.core.formatter.keep_record_constructor_on_one_line=one_line_never
+org.eclipse.jdt.core.formatter.keep_record_declaration_on_one_line=one_line_never
+org.eclipse.jdt.core.formatter.keep_simple_do_while_body_on_same_line=false
+org.eclipse.jdt.core.formatter.keep_simple_for_body_on_same_line=false
+org.eclipse.jdt.core.formatter.keep_simple_getter_setter_on_one_line=false
+org.eclipse.jdt.core.formatter.keep_simple_while_body_on_same_line=false
+org.eclipse.jdt.core.formatter.keep_switch_body_block_on_one_line=one_line_never
+org.eclipse.jdt.core.formatter.keep_switch_case_with_arrow_on_one_line=one_line_never
 org.eclipse.jdt.core.formatter.keep_then_statement_on_same_line=false
-org.eclipse.jdt.core.formatter.lineSplit=200
+org.eclipse.jdt.core.formatter.keep_type_declaration_on_one_line=one_line_never
+org.eclipse.jdt.core.formatter.lineSplit=120
```

Comment:
I would strongly prefer a column limit of 100 or less, for these reasons:

* Code reads better when it is constrained to a fairly small column limit as your eyes can simply scan down the page.
* 100 is already a generous limit, 120 starts to defeat the point of a column limit to begin with. Python's `black` only permits 88, 80 is the standard in the linux kernel, `cargo fmt` uses 100 by default, and google's java code style uses 100. To my knowledge, *no one* uses 120, at that point they just don't enforce a limit.
* Lets all code fit on the screen without the need to scroll right
* Important when editing from command line, either in `git diff` or in `vim`
* Important for those of us with only 1 monitor (like me 😢 )
* Important for viewing two files side by side, or having web browser + ide side by side
* Important for those that use vertical monitors (very common)

Reply:
> Important for those of us with only 1 monitor (like me 😢 )

Does your monitor only fit 80 charaters?

> Code reads better when it is constrained to a fairly small column limit as your eyes can simply scan down the page.

IMO, code reads better when you don't have 10 lines that are effectively one line of code. I agreed to line limit at all because you literally forcing me, but I disagree with this reasoning. In 95% of code reading you don't care for arguments of a function call, only the call itself. That's at least how I read code.

Reply:
I put 120 because .editroconfig has 120, and all point of this is to match it. You should've said this all in #162...

Reply:
I did say this in discord

Reply:
My monitor looks like this generally. I have marked out the 80, 100, and 20 character column limits on the vertical rulers so you can see them clearly. I have a fairly widescreen monitor mind you, and its 32 inches so its quite large. I believe this to be a best case scenario of someone with one monitor. Note that IDE annotations make this *even longer* than 120.

No other windows or code, everything fits on the screen (but don't disregard my statements about readability of vertical oriented code)
![image](https://user-images.githubusercontent.com/6969415/167467709-00fa9d5a-dae1-4eb8-a330-289b4180fc4f.png)
![image](https://user-images.githubusercontent.com/6969415/167468318-64c56de2-402d-487a-a257-bf54b52825a3.png)

From here on out, things don't fit at 120 characters. For example, side-by-side editing of code (extremely common, I reference flatbuffer schema while working on java).
![image](https://user-images.githubusercontent.com/6969415/167467871-dbc091b0-4894-46b4-8278-e36f3944e700.png)
![image](https://user-images.githubusercontent.com/6969415/167468553-96b199b2-ef1b-4c85-8564-e086318c43f0.png)

Github reviews/documentation reference/whatever else you use a web browser for.
![image](https://user-images.githubusercontent.com/6969415/167467957-e921dcef-a251-476d-b2f6-067147d080a4.png)
![image](https://user-images.githubusercontent.com/6969415/167468652-a6b87c7e-2939-4a22-9984-f3de21155b5c.png)

Maybe its my fault for not resizing my windows 🤔 Lets see how small I need to make my browser to get it to fit
![image](https://user-images.githubusercontent.com/6969415/167468059-70377420-2db5-45a5-bfe2-1ad7e9773903.png)
![image](https://user-images.githubusercontent.com/6969415/167468940-806baee4-e4ac-433d-a55b-878b0b15bc9f.png)

Reply:
I hope this demonstrated that the concerns I'm raising are not strange edge cases that no one runs into. I use these development workflows routinely (I never have my IDE in fullscreen mode, because I'm always looking at documentation, or on discord, or looking at github at the same time)